worker/jobTypes/githubJob.js

```diff
@@ -268,6 +309,34 @@ class GitHubJobClass {
                 'ERROR: makefile does not exist in /makefiles directory on meta branch.'
             );
         }
+        //set up env vars for all jobs
+        const pathPrefix = await this.writeEnvFile(isProdDeployJob)
+        
+        // server specifies path prefix for stagel commit jobs and prod deploy jobs only, which we
+        // save to job object to pass to mut in S3Publish.js. Front end constructs path for regular staging jobs 
+        // via the env vars defined/written in writeEnvProdFile, so the server doesn't have to create one here
+        if(typeof pathPrefix !== 'undefined' && pathPrefix !== null){
+          this.currentJob.payload.pathPrefix = pathPrefix;
+        }
+        
+        // default commands to run to build repo
+        const commandsToBuild = [
+          `. /venv/bin/activate`,
+          `cd repos/${this.getRepoDirName()}`,
+          `rm -f makefile`,
+          `make html`
+      ];
+
+      // check if need to build next-gen
+      if (this.buildNextGen() && !isProdDeployJob) {
+        commandsToBuild[commandsToBuild.length - 1] = 'make next-gen-html';
+      }
+
+      //check if prod deploy job
+      if (isProdDeployJob) {
+          commandsToBuild[commandsToBuild.length - 1] = 'make download-published-branches';
```

Comment:
This target is very specific in name -- can we call it something like get-build-dependencies?